### PR TITLE
add Dockerfile

### PR DESCRIPTION
**Pull request by 0xBruno**
Dockerfile for running gcpwn from docker container. 

Build with (inside repo with Dockerfile): 
`docker build -t gcpwn .`

Run with:  
` docker run -it -v ~/.config/gcloud:/root/.config/gcloud gcpwn `

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM ubuntu:latest
+
+RUN apt-get update -y
+RUN apt-get install apt-transport-https ca-certificates gnupg curl python3 python3-pip -y
```
Comment (WebbinRoot) [inline Dockerfile:4]: Can change to "apt-get install apt-transport-https ca-certificates gnupg curl python3 python3-venv -y"

Can apt install python3-venv early on, and don't need to apt install pip it looks like with virtual environments.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --dearmor -o /usr/share/keyrings/cloud.google.gpg
+
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" >> /etc/apt/sources.list.d/google-cloud-sdk.list
+
+RUN apt-get update -y && apt-get install google-cloud-cli -y
+
+WORKDIR /usr/src/gcpwn
+
+ADD . ./
+RUN apt-get install python3.12-venv -y
```
Comment (WebbinRoot) [inline Dockerfile:15]: Can delete line per comment above

**Comment by WebbinRoot** [comment]
Added some comments above for just some apt install management. 

On separate note, were you able to add credentials running the tool from the docker container? It seems like trying to add service credentials, for example, doesn't seem to work with the docker container as I think its expecting the credentials to be within the docker container? Maybe its a runtime issue on my end but curious what that looks like for you to load in key.json for example. It might be easier to just set up the entry point as bin/bash in the user home folder to make it a container for dropping in creds as well.

**Comment by 0xBruno** [comment]
Changed the docker run command to load the adc creds in with: 

` docker run -it -v ~/.config/gcloud:/root/.config/gcloud gcpwn`

**Comment by WebbinRoot** [comment]
One item I noticed is GatheredData folder populates to the local folder within the docker container so when you try to break out of the tool to view it it exits the container and wipes the data. For example, load in a service account key for credentials and run `modules run enum_buckets --download` with files in buckets. 

1. Changed ADD . ./ to COPY . ./
2. Specific python env to just "python3-venv" 
3. Different entrypoint to drop into a shell already in necessary python env. This way you drop into a container and can see the data as you download anything/gen CSVs, and you can still choose to mount gcloud config if you want for the gcloud auth. 

Let me know if this works on your end and I can merge after making changes.

```
FROM ubuntu:latest

RUN apt-get update -y
RUN apt-get install apt-transport-https ca-certificates gnupg curl python3 python3-venv -y

RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --dearmor -o /usr/share/keyrings/cloud.google.gpg
RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" >> /etc/apt/sources.list.d/google-cloud-sdk.list

RUN apt-get update -y &&  apt-get install google-cloud-cli -y

WORKDIR /usr/src/gcpwn

COPY . ./
RUN python3 -m venv venv
RUN  ./venv/bin/pip install -r /usr/src/gcpwn/requirements.txt

ENTRYPOINT ["/bin/bash","-c","source /usr/src/gcpwn/venv/bin/activate && exec /bin/bash"]
```

**Comment by 0xBruno** [comment]
That works. Another option I tested was using a volume to GatheredData and reviewing the data on your host system but gets a bit annoying. 

e.g.
`VOLUME ["/usr/src/gcpwn/GatheredData"]`

**Pull request merged**